### DISCUSSION TITLE
LayoutOptions.EndAndExpand is obsolete

### DISCUSSION THREAD

**Discussion by devonuto** [dotnet/maui#7346] [Q&A] [answered]
So I have a question.  I have two grids inside a stacklayout. I want the second grid to sit at the end of the page, so I use EndAndExpand, but I get this:

LayoutOptions.EndAndExpand is obsolete: 'The StackLayout expansion options are deprecated; please use a Grid instead.'

However, my question is, how do you tell the stacklayout to be fullscreen, so just using LayoutOptions.End will have the grid at the bottom of the page? Nothing except this so-called "obsolete" call works that I can find, without having to define a new grid to sit these grids inside, which is not ideal (due to styling applied on grids around margins, etc).

**Comment by hartez** [comment, marked as answer]
The "AndExpand" options from Forms were removed from .NET MAUI for performance and clarity reasons.

Your options are 

- ignore the obsolete warning and use the legacy "AndExpand" support on StackLayout
- define a grid which contains your other grids and apply a different style to it
- create a [custom layout class which provides a single-column fullscreen layout](https://github.com/hartez/CustomLayoutExamples#single-column-layout)

**Reply by devonuto** [reply]
Thanks, I'll have a look at that layout class

**Comment by Divyesh-Bhatt** [comment]
@hartez this deprecation is only for StackLayout? or it applies to all other controls as well like Button, Label, Entry etc?

**Reply by hartez** [reply]
The options are deprecated for all controls. But `EndAndExpand`, `CenterAndExpand`, `StartAndExpand`, and `FillAndExpand` only affect how a control is laid out inside of a `StackLayout`. So this change has no effect on other types of layouts.

**Comment by ToolmakerSteve** [comment]
All controls. In Xamarin.Forms, It did nothing anyway, if parent was not a StackLayout. Expand was not meaningful inside any other layout. They each have their own way of controlling size.
